packages/ra-core/src/sideEffect/useNotify.ts

```diff
@@ -28,13 +28,15 @@ const useNotify = () => {
             type: NotificationType = 'info',
             messageArgs: any = {},
             undoable: boolean = false,
-            autoHideDuration?: number
+            autoHideDuration?: number,
+            multiLine?: boolean
```

Comment:
At this stage, it becomes obvious that the signature of the `notify` callback isn't right. As we'r adding new options, it makes calls super weird. For instance, to enable multiline without changing other parameters:

```jsx
notify('foo', undefined, undefined, undefined, undefined, true)
```

We must switch to a simpler signature: `notify(message, options)`. This can be done in a backwards-compatible way by inspecting the type of the second argument at runtime. TypeScript may get in the way, but it supports multiple call signatures.